### PR TITLE
Update of dev environment

### PR DESCRIPTION
**Pull request by evdherberg**
Containing:
- Unified structure for auto-loading plugins for different modes of operation
- Added in heapdump/monitoring/jsxc plugins for auto-loading
- Modified scripts to work with (docker compose 2) default of using hyphens for creating service names
- Fixed copy monitoring plugin script and added copy hazelcast plugin script
- Extra scripts for blocking and unblocking server 2 in federation mode

### CODE REVIEW

**Comment by evdherberg** [comment]
And yeah, this includes binary blobs. Maybe not the best thing. But very convenient. I can take it out if necessary.

File: scripts/copy_hazelcast_plugin.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+# For when you've built the hazelcast plugin locally and want this in your next stack
+cp ../openfire-hazelcast-plugin/target/hazelcast-openfire-plugin-assembly.jar ./plugins_for_clustered/hazelcast.jar
+cp ../openfire-hazelcast-plugin/target/hazelcast-openfire-plugin-assembly.jar ./plugins_for_federated/hazelcast.jar
+cp ../openfire-hazelcast-plugin/target/hazelcast-openfire-plugin-assembly.jar ./plugins_for_other/hazelcast.jar
```
Comment (Fishbowler) [inline scripts/copy_hazelcast_plugin.sh:5]: Do these folders want the Hazelcast plugin?

File: docker-compose-federated.yml [hunk 2]
```diff
@@ -65,6 +66,7 @@ services:
       - "db2"
     volumes:
       - ./_data/xmpp/federated/2/conf:/var/lib/openfire/conf
+      - ./_data/plugins_for_federated:/opt/plugins
```
Comment (Fishbowler) [inline docker-compose-federated.yml:69]: Awesome idea!

**Review by Fishbowler — COMMENTED** [review]
Will we always want the Monitoring plugin in a given stack? Adding an optional BOSH client is unobstructive, but adding a plugin that provides additional capabilities could get in the way.

How do you feel about including the script but not the blobs? Would that have the effect of "off by default"?

File: scripts/copy_hazelcast_plugin.sh (hunk 1 above)
Comment (evdherberg) [reply scripts/copy_hazelcast_plugin.sh]: Fair points. In the latest commit I removed two incorrect copy lines, and I removed the monitoring plugin binary from the default sets.

**Pull request merged**